server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -462,6 +462,15 @@ protected static void parseCompositeField(
                     )
                 );
             }
+            if (IndexMetadata.INDEX_APPEND_ONLY_ENABLED_SETTING.get(parserContext.getSettings()) == false) {
```

Comment:
This will be breaking change but as Star tree is marked experimental, this should be fine.